tests/matlab_test_data_collectors/matlab_collectors/collect_makeCartCircle.m

```diff
@@ -2,21 +2,24 @@
 all_params = {
     {5, 100, [1,1],  2*pi}, ...
     {1, 10, [5,5],  pi/4}, ...
-    {0.5, 10, [0,0],  2*pi}
+    {0.5, 10, [0,0],  2*pi}, ...
+    {0.05, 20, [0 0]}
     };  
-output_folder = 'collectedValues/makeCartCircle/';
+output_file = 'collectedValues/makeCartCircle.mat';
+recorder = utils.TestRecorder(output_file);
 
 for idx=1:length(all_params)
-        circle = makeCartCircle(all_params{idx}{1}, all_params{idx}{2}, all_params{idx}{3}, all_params{idx}{4});
         params = all_params{idx};
-        idx_padded = sprintf('%06d', idx - 1);
-
-        idx_padded = sprintf('%06d', idx - 1);
-        if ~exist(output_folder, 'dir')
-            mkdir(output_folder);
+        if length(all_params{idx}) < 4
+            circle = makeCartCircle(params{1}, params{2}, params{3});
+            params = {params{1}, params{2}, params{3}};
+        else
+            circle = makeCartCircle(params{1}, params{2}, params{3}, params{4});
+            params = {params{1}, params{2}, params{3}, params{4}};
         end
-        filename = [output_folder idx_padded '.mat'];
-        save(filename, 'params', 'circle');
+        recorder.recordVariable('params', params);
+        recorder.recordVariable('circle', circle);
+        recorder.increment()
 
 end
-disp('Done.')
+recorder.saveRecordsToDisk();
```

Comment:
So we did not save outputs before? :hushed:

Reply:
nope!